src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/Regex.Match.Tests.cs

```diff
@@ -1253,9 +1253,10 @@ public void Match_InstanceMethods_DefaultTimeout_Throws(RegexEngine engine)
                 const string Pattern = @"^([0-9a-zA-Z]([-.\w]*[0-9a-zA-Z])*@(([0-9a-zA-Z])+([-\w]*[0-9a-zA-Z])*\.)+[a-zA-Z]{2,9})$";
                 string input = new string('a', 50) + "@a.a";
 
-                AppDomain.CurrentDomain.SetData(RegexHelpers.DefaultMatchTimeout_ConfigKeyName, TimeSpan.FromMilliseconds(100));
-
                 Regex r = await RegexHelpers.GetRegexAsync(engine, Pattern);
+
+                // Generating the regex above itself uses a regex internally, so set the short timeout we want only after that's done
+                AppDomain.CurrentDomain.SetData(RegexHelpers.DefaultMatchTimeout_ConfigKeyName, TimeSpan.FromMilliseconds(100));
```

Comment:
This isn't valid _because_ GetRegexAsync uses regexes.  Once it does, the timeout will be read once from the appcontext data and cached, such that overwriting it won't help.

Reply:
Yes. Any suggestion?

Reply:
Private reflection?

Reply:
Hmm, it seemed to me that for the generated regex, I could put the timeout set after the regex is generated, because it's read inside the generated code. But it still hangs - what am I missing.

Reply:
All generated regexes share the same timeout field. Has _any_ been accessed?